cmd/lncli/commands.go

```diff
@@ -4016,3 +4030,255 @@ func restoreChanBackup(ctx *cli.Context) error {
 
 	return nil
 }
+
+var newMacaroonCommand = cli.Command{
+	Name:     "newmacaroon",
+	Category: "Macaroons",
+	Usage: "Bakes a new macaroon with the provided list of " +
+		"permissions and restrictions",
+	ArgsUsage: "--permission=entity/action [--format= | --save_to=] " +
+		"[--timeout=] [--ip_address=]",
+	Description: `
+	Create a new macaroon that grants the provided permissions and
+	optionally adds restrictions (timeout, IP address) to it.
+
+	The new macaroon can be shown on command line either in JSON
+	(--format=json) or hex (--format=hex) serialized format.
+	Or it can be saved directly to a file using the --save_to argument.
+
+	A permission is a tuple of an entity and an action.
+	The following entities are available in lnd:
```

Comment:
This can quickly become outdated. Perhaps it's worth introducing a flag/separate command to list the entities and actions available? Exposing which ones all available RPCs require would be nice too, at the moment users have to read the code. Sounds like follow-up work to me though.